app/scripts/modules/core/src/managed/ArtifactDetail.tsx

```diff
@@ -34,6 +34,12 @@ import { isResourceKindSupported } from './resources/resourceRegistry';
 
 import './ArtifactDetail.less';
 
+const SUPPORTED_PRE_DEPLOYMENT_TYPES = [
```

Comment:
i moved this from `PreDeploymentStepCard.tsx` because it fixes another bug which was a non-issue before this change, but became one after. when there's unsupported types in the list of steps, we were unable to know from this file whether or not all the steps taken in by `PreDeploymentStepCard` would end up rendering anything. That meant that in versions with _only_ a `BUILD` event, we'd still show the "PRE-DEPLOYMENT" section but the card component wouldn't render anything underneath.

With the filtering on L256, we'll avoid showing an empty section.